adapter/gorm/logger.go

```diff
@@ -0,0 +1,58 @@
+package gorm
+
+import (
+	"context"
+	"time"
+
+	coopLogger "github.com/coopnorge/go-logger"
+	"gorm.io/gorm/logger"
+)
+
+type gormLogger struct {
+	instance *coopLogger.Logger
+}
+
+// NewLogger creates a new Gorm logger that passes message to go-logger
+func NewLogger(opts ...LoggerOption) logger.Interface {
```

Comment:
"Accept Interfaces, Return Concrete Types". can we stick to this principle please?

Reply:
I'll test if Gorm is will to accept that, this was taken from Gorm sample logger.

Reply:
Gorm was happy, our linter setup was not.

```shell
```

Reply:
Then why not export the type? Is there a reason why the type has to be price which it is being returned?

Reply:
We are also forced to break "Accept Interfaces, Return Concrete Types" here.

```go
func (l *gormLogger) LogMode(logger.LogLevel) logger.Interface {
	return l
}
```

As this is part of the interface we are implementing.

Reply:
Should we export a type that we do not want people to instantiate directly to satisfy the linter?

I do not yet have strong feelings about this in either direction, this can simmer for a bit.

Reply:
Yah, but that's due to the restriction of the interface we are implementing correct?

Just put a:

```
var _ logger.Interface = &GormLogger{}
```

This will make sure that the function LogMode returns an interface because that's what the interface `GormLogger` is implementing wants.

Also, in regards to returning exported types that aren't supposed to be directly instantiated, looks like stdlibs also do that.

https://github.com/golang/go/blob/master/src/os/types.go#L18-L20

Here, file is not supposed to be instantiated on it's own, but it's exported anyways.

Reply:
nice with some precedence :)

Reply:
fixed

Reply:
We should look into catching this using a linter.